File: LeetCode_566_2d_vector.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        if(nums[0].size()*nums.size() != r*c)
            return nums;

        vector<vector<int>> tmp(r);
        for(int i = 0;i < tmp.size(); i++)
        {
            tmp[i].resize(c);
        }

        int it_pos = 0;
        vector<vector<int>>::iterator it_x = nums.begin();
        vector<int>::iterator it = nums[it_pos].begin();

        for(int i = 0;i < r;i++)
            for(int j = 0; j < c;j++)
            {
                tmp[i][j] = *it;
                it++;
                if(it == nums[it_pos].end())
                {
                    it_pos++;
                    it = nums[it_pos].begin();
                }

            }
        return tmp;
    }
};

File: leetcode_archived_cpp/LeetCode_566_2d_vector.py
# Method 1
# class Solution:
#     def matrixReshape(self, nums, r, c):
#         """
#         :type nums: List[List[int]]
#         :type r: int
#         :type c: int
#         :rtype: List[List[int]]
#         """
#         if (len(nums[0])*len(nums)) != r*c:
#             return nums
#          #tmp = [[None] * c for i in range(r)]
#         tmp = []
#         tmp_col = []
#         count = 0
#         for i in nums:
#             for j in i:
#                 tmp_col.append(j)
#                 count += 1
#                 if count == c:
#                     count = 0
#                     tmp.append(tmp_col)
#                     tmp_col = []
#         return tmp

# Method 2
# import numpy as np
# class Solution:
#     def matrixReshape(self, nums, r, c):
#         """
#         :type nums: List[List[int]]
#         :type r: int
#         :type c: int
#         :rtype: List[List[int]]
#         """
#         try:
#             return np.reshape(nums,(r,c)).tolist()
#         except:
#             return nums

# Method 3
# bug: list out of range……
